tests/xtimer_usleep/main.c

```diff
@@ -24,35 +26,77 @@
 
 #include "xtimer.h"
 #include "timex.h"
+#include "stdlib.h"
 
 #define RUNS                (5U)
 #define SLEEP_TIMES_NUMOF   (sizeof(sleep_times) / sizeof(sleep_times[0]))
 
-static const uint32_t sleep_times[] = { 10000, 50000, 100000 };
+static const uint32_t sleep_times[] = { 10000, 50000, 1234, 5678, 121212, 98765, 75000};
 
+#define ERROR_US 500
+
+/*
+ * To use a pin to probe the sleep times enable the gpio module and define
+ * respective SLEEP_GPIO_PIN and set SLEEP_PIN to (1)
+ * */
+
+#define SLEEP_PIN (0)
+#if SLEEP_PIN
+#include "board.h"
+#include "periph/gpio.h"
+#define SLEEP_GPIO_PIN  GPIO_PIN(PORT_F, 7)
+#endif
 int main(void)
 {
     uint32_t start_test, testtime;
+    uint32_t start_sleep;
+    uint32_t diff;
+    int32_t err;
+
+#if SLEEP_PIN
+    gpio_init( SLEEP_GPIO_PIN, GPIO_OUT );
+#endif
 
     printf("Running test %u times with %u distinct sleep times\n", RUNS,
            (unsigned)SLEEP_TIMES_NUMOF);
-    puts("Please hit any key and then ENTER to continue");
-    getchar();
+    puts("Please hit any key and then ENTER to continue\n");
+    // getchar();
     start_test = xtimer_now_usec();
     for (unsigned m = 0; m < RUNS; m++) {
         for (unsigned n = 0;
              n < sizeof(sleep_times) / sizeof(sleep_times[0]);
              n++) {
-            uint32_t diff, start;
-            start = xtimer_now_usec();
+
+            start_sleep = xtimer_now_usec();
```

Comment:
Change the scope of your variables, they might be defined only in the context they're used.